File: app/src/main/java/com/github/abhijitpparate/keeps/screen/note/presenter/NoteContract.java
package com.github.abhijitpparate.keeps.screen.note.presenter;


import android.support.annotation.ColorRes;
import android.support.annotation.StringRes;

import com.github.abhijitpparate.checklistview.CheckListItem;
import com.github.abhijitpparate.keeps.data.database.Note;

import java.util.List;

public interface NoteContract {

    public enum NoteColor {
        DEFAULT, WHITE, PURPLE, RED, GREEN, YELLOW, BLUE, ORANGE
    }

    interface View {

        String getNoteTitle();
        String getNoteBody();
        String getCheckList();

        void setNote(Note notes);
        void setNoteTitle(String title);
        void setNoteBody(String body);
        void setNoteChecklist(List<CheckListItem> checklist);
        void setNoteColor(@ColorRes int color);

        void showOptionsPanel();
        void hideOptionsPanel();
        void showHomeScreen();

        void loadNoteIfAvailable();

        void setPresenter(NoteContract.Presenter presenter);
        void makeToast(@StringRes int strId);
        void makeToast(String message);

        void switchToChecklist();
        void switchToText();
    }

    interface Presenter {
        void onSaveClick();

        void loadNote(String noteId);

        void subscribe();
        void unsubscribe();

        void onChecklistClick(boolean isChecked);
        void onDrawingClick();
        void onAudioClick();
        void onVideoClick();
        void onImageClick();

        void onOptionsClick();
        void onDeleteClick();
        void onSendClick();
        void onDuplicateClick();
        void onLabelClick();
        void onColorSelected(NoteColor color);
    }
}

File: app/src/main/java/com/github/abhijitpparate/keeps/screen/note/presenter/NoteContract.kt
package com.github.abhijitpparate.keeps.screen.note.presenter


import android.support.annotation.ColorRes
import android.support.annotation.StringRes

import com.github.abhijitpparate.checklistview.CheckListItem
import com.github.abhijitpparate.keeps.data.database.Note

interface NoteContract {

    enum class NoteColor {
        DEFAULT, WHITE, PURPLE, RED, GREEN, YELLOW, BLUE, ORANGE
    }

    interface View {

        val noteId: String
        var noteTitle: String
        var noteBody: String
        val checkList: String
        val noteColor: String

        fun setNote(notes: Note)
        fun setNoteChecklist(checklist: List<CheckListItem>)
        fun setNoteColor(@ColorRes colorInt: Int, colorString: String)

        fun showOptionsPanel()
        fun hideOptionsPanel()
        fun showHomeScreen()

        fun loadNoteIfAvailable()

        fun setPresenter(presenter: NoteContract.Presenter)
        fun showProgressbar(bool: Boolean)
        fun makeToast(@StringRes strId: Int)
        fun makeToast(message: String)

        fun switchToChecklist()
        fun switchToText()
    }

    interface Presenter {
        fun onSaveClick()

        fun loadNote(noteId: String)

        fun subscribe()
        fun unsubscribe()

        fun onChecklistClick(isChecked: Boolean)
        fun onDrawingClick()
        fun onAudioClick()
        fun onVideoClick()
        fun onImageClick()

        fun onOptionsClick()
        fun onDeleteClick()
        fun onSendClick()
        fun onDuplicateClick()
        fun onLabelClick()
        fun onColorSelected(color: NoteColor)
    }
}
